search/includes/classes/class-search.php

```diff
@@ -1624,6 +1635,57 @@ public function filter__ep_prepare_meta_allowed_protected_keys( $keys, $post ) {
 		return array_merge( $keys, $vip_search_allow_list_keys );
 	}
 
+	/**
+	 * Hooks into the ep_$indexable_mapping hooks to add things like allocation rules
+	 * 
+	 * Note that this hook receives the mapping and settings together
+	 */
+	public function filter__ep_indexable_mapping( $mapping ) {
+		if ( ! is_array( $mapping['settings'] ) ) {
+			return $mapping;
+		}
+
+		$origin_datacenter = $this->get_index_routing_allocation_include_dc();
+
+		if ( $origin_datacenter ) {
+			// We want all indexes to live in the site's origin datacenter
+			$mapping['settings']['index.routing.allocation.include.dc'] = $origin_datacenter;
+		}
+
+		return $mapping;
+	}
+
+	public function get_index_routing_allocation_include_dc() {
+		$dc = defined( 'VIP_ORIGIN_DATACENTER' ) ? VIP_ORIGIN_DATACENTER : $this->get_origin_dc_from_es_endpoint( $this->get_current_host() );
```

Comment:
```suggestion
		$dc = strtolower( defined( 'VIP_ORIGIN_DATACENTER' ) ? VIP_ORIGIN_DATACENTER : $this->get_origin_dc_from_es_endpoint( $this->get_current_host() ) );
```

Just to be extra safe here, we do usually refer to them in ALLCAPS, so it's not inconceivable that `VIP_ORIGIN_DATACENTER` might get set to ALLCAPS due to a mistake/misconfiguration

Reply:
Good catch - I've updated this in 28b59c1fe (went with a slightly different approach for readability)